src/common/cli_wrapper.cpp

```diff
@@ -82,21 +81,14 @@ void CLIWrapper::switchGroup(const std::string &name) {
     currentGroup_ = name;
 }
 
-void CLIWrapper::parse(int argc, char** argv) {
+YAML::Node CLIWrapper::parse(int argc, char **argv) {
```

Comment:
Now I am confused. Does `parse()` perform a state change to the CLIWrapper class, or is it a pure function that creates a YAML::Node object?

It seems it does both: It updates config_, and then returns it. I feel this is incorrect design.

Reply:
One possible solution is to make this `void` again, but then add an automatic type cast for this class to YAML::Node. Then you can say

    clip.parse(...);
    YAML::Node opts = cli;

Reply:
I agree and I am making it void again. 

I'll also add a @TODO here. We need to clean the entire Options/Config/CLI chaos anyway (which is one of the next steps). If we spend to much time here now, will not go forward.

Reply:
Went back to void and removed the get/set functions for the YAML tree, this now can only be passed in the constructor as a reference. Hence proper wrapper.